sssd_test_framework/roles/ipa.py

```diff
@@ -2449,3 +2722,787 @@ def print(self) -> str:
+            "host": host,
+            "service": service,
+        }
+
+    def status(
+        self,
+        *,
+        user: str | None = None,
+        group: str | None = None,
+        host: str | None = None,
+        service: str | None = None,
+        include_members: bool = False,
+    ) -> dict[str, Any]:
+        """
+        Get rule status, optionally checking membership and returning the raw member lists.
+
+        :param user: Username to check for membership in the rule.
+        :param group: Group name to check for membership in the rule.
+        :param host: Hostname to check for membership in the rule.
+        :param service: Service name to check for membership in the rule.
+        :param include_members: When ``True`` return the resolved member lists in the output.
+        :return: Dictionary with rule status information and optional membership results.
+        :rtype: dict[str, Any]
+        """
+        raw_data = self._fetch_rule_data()
+
+        users = to_list(get_attr(raw_data, "Users", default=[]))
+        user_groups = to_list(get_attr(raw_data, "User Groups", default=[]))
+        hosts = to_list(get_attr(raw_data, "Hosts", default=[]))
+        host_groups = to_list(get_attr(raw_data, "Host Groups", default=[]))
+        services = to_list(get_attr(raw_data, "Services", default=[]))
+        service_groups = to_list(get_attr(raw_data, "Service Groups", default=[]))
+
+        membership_checks: dict[str, bool] = {}
+        if user is not None:
+            membership_checks["user"] = user in users
+        if group is not None:
+            membership_checks["group"] = group in user_groups
+        if host is not None:
+            membership_checks["host"] = host in hosts
+        if service is not None:
+            membership_checks["service"] = service in services
+
+        payload: dict[str, Any] = {
+            "name": self.name,
+            "enabled": str(get_attr(raw_data, "Enabled", default="FALSE")).upper() == "TRUE",
+            "description": get_attr(raw_data, "Description"),
+            "user_count": len(users) + len(user_groups),
+            "host_count": len(hosts) + len(host_groups),
+            "service_count": len(services) + len(service_groups),
+            "categories": {
+                "user": get_attr(raw_data, "User category"),
+                "host": get_attr(raw_data, "Host category"),
+                "service": get_attr(raw_data, "Service category"),
+            },
+        }
+
+        if membership_checks:
+            payload["membership"] = membership_checks
+
+        if include_members:
+            payload["members"] = {
+                "users": users,
+                "user_groups": user_groups,
+                "hosts": hosts,
+                "host_groups": host_groups,
+                "services": services,
+                "service_groups": service_groups,
+            }
+
+        return payload
+
+    def contains(self, **kwargs) -> bool:
```

Comment:
How will this be used?

Reply:
Something like this, I was thinking...
```
    ssh_rule = ipa.hbac("ssh_access_test").create(
        users=["testuser"],
        hosts=["host.test"],
        services=["sshd"]
    )

    assert ssh_rule.contains(user="testuser")
    assert ssh_rule.contains(host="host.test")
    assert not ssh_rule.contains(user="otheruser")
```
`contains`  boolean membership checks in an IPA HBAC context 

Suggestions are welcome. Should we keep it as it is, or do you recommend updating it?

Reply:
Okay no problem for me. I was just curious.